src/metabase/email/messages.clj

```diff
@@ -57,40 +57,46 @@
 
 ;; HACK: temporary workaround to postal requiring a file as the attachment
 (defn- write-byte-array-to-temp-file
-  [img-bytes]
-  (let [file (java.io.File/createTempFile "metabase_pulse_image_" ".png")
-        fos (new java.io.FileOutputStream file)]
-    (.deleteOnExit file)
-    (.write fos img-bytes)
-    (.close fos)
+  [^bytes img-bytes]
+  (let [file (doto (java.io.File/createTempFile "metabase_pulse_image_" ".png")
+               .deleteOnExit)]
+    (with-open [fos (java.io.FileOutputStream. file)]
+      (.write fos img-bytes))
     file))
 
 (defn- find-byte-array
-  [needle haystack]
-  (first (keep-indexed #(when (java.util.Arrays/equals %2 needle) %1) haystack)))
+  [^bytes needle, haystack]
+  (first (keep (fn [^bytes item]
+                 (when (java.util.Arrays/equals item needle)
+                   true))
+               haystack)))
+
+(defn- render-image [images-atom bytes]
+  (let [index (or (find-byte-array bytes @images-atom)
+                  (count @images-atom))]
+    (when (= index (count @images-atom))
+      (swap! images-atom conj bytes))
+    (str "cid:IMAGE" index)))
 
 (defn render-pulse-email
   "Take a pulse object and list of results, returns an array of attachment objects for an email"
   [pulse results]
   (let [images       (atom [])
-        render-img   (fn [bytes]
-                        (let [index (or (find-byte-array bytes @images) (count @images))]
-                          (if (= index (count @images)) (reset! images (conj @images bytes)))
-                          (str "cid:IMAGE" index)))
-        body         (apply vector :div (mapv (partial render-pulse-section render-img true) results))
+        body         (apply vector :div (for [result results]
+                                          (render-pulse-section (partial render-image images) :include-buttons result)))
```

Comment:
one other thing I forgot. It's more readable to use a keyword for boolean params
```clojure
(render-pulse-section (partial render-image images) :include-buttons result)
```

instead of 
```clojure
(render-pulse-section (partial render-image images) true result)
```

I'd even say you should do
```clojure
(render-pulse-section (partial render-image images) (not :include-buttons) result)
```

Instead of 
```clojure
(render-pulse-section (partial render-image images) false result)
```